src/sentry/models/event.py

```diff
@@ -57,6 +58,10 @@ class Meta:
 
     __repr__ = sane_repr('project_id', 'group_id')
 
+    @classmethod
+    def get_node_id(cls, project_id, event_id):
```

Comment:
This sort of implies that it's used to also get the node ID for lookup, not just creation since the node ID is then saved as the field data in the database — it seems like an easy mistake to make/misconception to have since this code is pretty confusing already. I don't have any concrete suggestions, but there might be a better name for this that avoids confusion?